packages/jest-resolve/src/node_modules_paths.js

```diff
@@ -37,11 +38,17 @@ export default function nodeModulesPaths(
     prefix = '\\\\';
   }
 
-  const paths = [basedirAbs];
-  let parsed = path.parse(basedirAbs);
+  // The node resolution algorithm (as implemented by NodeJS
+  // and TypeScript) traverses parents of the physical path,
+  // not the symlinked path 
+  const physicalBasedir: string = fs.realpathSync(basedirAbs);
+
+  const paths = [physicalBasedir];
+  let parsed = path.parse(physicalBasedir);
   while (parsed.dir !== paths[paths.length - 1]) {
-    paths.push(parsed.dir);
-    parsed = path.parse(parsed.dir);
+    const realParsedDir: string = fs.realpathSync(parsed.dir);
```

Comment:
I don't think you need the type annotation here.

Reply:
Ah, yes, we require this in our repositories (even in some places where it is a bit redundant). I'll go ahead and remove it.

Reply:
```typescript
// obviously x is a number, so this annotation is redundant
const x = 123;
```

```typescript
// Nobody has any clue what the type of "x" is without
// using an analysis tool like VS Code
const x = getThing(); 
```